src/ciadmin/generate/worker_pools.py

```diff
@@ -207,7 +207,11 @@ def get_azure_provider_config(
         for vmSize in vmSizes:
             loc = location.replace("-", "")
             DeploymentId = image.image_id(provider_id, "deployment_id")
-            ImageId = image.image_id(provider_id, location)
+            try:
+                version = image.image_id(provider_id, "version")
+            except Exception:
+                version = None
+            DeploymentId = image.image_id(provider_id, "deployment_id")
```

Comment:
It looks like there's now two copies of this line? One of them probably ought to be removed.